### PR TITLE
Build by strategy

### PR DESCRIPTION
**Pull request by kwk**
This adds a workflow option to select which build-strategy to build for (e.g. big-merge or pgo at the moment.)

### CODE REVIEW

File: .github/workflows/fedora-copr-build.yml [hunk 1]
```diff
@@ -30,6 +38,7 @@ jobs:
     container: fedora:39
     steps:
       - name: Setup Copr config file
+        if: github.event_name != 'workflow_dispatch' || (inputs.strategy == matrix.name)
```
[the remark this answers is not in the document]
Comment (kwk) [reply .github/workflows/fedora-copr-build.yml:41]: It actually is configurable. I think one needs to pass in a payload for the matrix and remember that I have done it before. Give some time.

**Comment by kwk** [comment]
@nikic I've implemented a dynamic matrix and tested it in some other repo.

File: .github/workflows/fedora-copr-build.yml [hunk 2]
```diff
@@ -6,26 +6,64 @@ on:
...
+      strategy:
+        description: 'build strategy'
+        required: true
+        type: choice
+        options:
+        - big-merge
+        - pgo
 
 jobs:
+  job1:
```
Comment (nikic) [inline .github/workflows/fedora-copr-build.yml:20]: job1 -> generate-matrix or something?

File: .github/workflows/fedora-copr-build.yml [hunk 3]
```diff
@@ -6,26 +6,64 @@ on:
...
+            "copr_target_project": "@fedora-llvm-team/llvm-snapshots-pgo",
+            "extra_script_file": "scripts/functions-pgo.sh",
+            "clone_url_tpl": "https://src.fedoraproject.org/forks/kkleine/rpms/PKG.git",
+            "clone_ref": "pgo"
+          }
+        EOF
+        )
+
+        if [[ '${{ inputs.strategy }}' == 'big-merge' ]]; then
+          NAMES='["big-merge"]'
```
Comment (nikic) [inline .github/workflows/fedora-copr-build.yml:51]: Per https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#example-adding-configurations I think just includes would be enough and we don't need to enumerate names separately?

Comment (kwk) [reply .github/workflows/fedora-copr-build.yml]: Addressed in 25a16dd

File: .github/workflows/fedora-copr-build.yml (hunk 2 above)
Comment (kwk) [reply .github/workflows/fedora-copr-build.yml]: Addressed in 25a16dd

**Comment by kwk** [comment]
@nikic ready for another review?

**Review by nikic — APPROVED** [review]

**Pull request merged**